src/sys/yanix/fs/dir.rs

```diff
@@ -69,27 +60,31 @@ impl Dir {
                 oflags |= OFlag::CREAT;
             }
         }
+        oflags |= OFlag::from_bits(options.ext.custom_flags).expect("unrecognized OFlag bits")
+            & !OFlag::ACCMODE;
         unsafe {
             let fd = openat(
                 self.std_file.as_raw_fd(),
                 path,
                 oflags,
-                Mode::from_bits(0o666).unwrap(),
+                #[allow(clippy::useless_conversion)]
+                Mode::from_bits(options.ext.mode.try_into().expect("unrecognized Mode bits"))
+                    .expect("unrecognized Mode bits"),
```

Comment:
Should we panic here? Or is a situation where the user specifies wrong mode bits possible, and which they'd like to recover from?

Reply:
Looks like libstd silently ignores unrecognized mode bits. I'll update to match.

Reply:
Done!